docs/csharp/programming-guide/classes-and-structs/polymorphism.md

```diff
@@ -1,93 +1,108 @@
+
+[!code-csharp[Basic hierarchy](~/samples/snippets/csharp/objectoriented/Hierarchy.cs#FirstHierarchy)]
+
+A derived class can stop virtual inheritance by declaring an override as [sealed](../../language-reference/keywords/sealed.md). Stopping inheritance requires putting the `sealed` keyword before the `override` keyword in the class member declaration. The following code provides an example:
+
+[!code-csharp[A sealed overridden member](~/samples/snippets/csharp/objectoriented/Hierarchy.cs#SealedOverride)]
+
+In the previous example, the method `DoWork` is no longer virtual to any class derived from `C`. It's still virtual for instances of `C`, even if they're cast to type `B` or type `A`. Sealed methods can be replaced by derived classes by using the `new` keyword, as the following example shows:
+
+[!code-csharp[New method declaration](~/samples/snippets/csharp/objectoriented/Hierarchy.cs#NewDeclaration)]
+
+In this case, if `DoWork` is called on `D` using a variable of type `D`, the new `DoWork` is called. If a variable of type `C`, `B`, or `A` is used to access an instance of `D`, a call to `DoWork` will follow the rules of virtual inheritance, routing those calls to the implementation of `DoWork` on class `C`.
+
+### Access base class virtual members from derived classes
+
+A derived class that has replaced or overridden a method or property can still access the method or property on the base class using the `base` keyword. The following code provides an example:
+
+```csharp
+public class Base
+{
+    public virtual void DoWork() {/*...*/ }
+}
+public class Derived : Base
+{
+    public override void DoWork()
+    {
+        //Perform Derived's work here
+        //...
+        // Call DoWork on base class
+        base.DoWork();
+    }
+}
+```
+
+For more information, see [base](../../language-reference/keywords/base.md).
+
 > [!NOTE]
-> It is recommended that virtual members use `base` to call the base class implementation of that member in their own implementation. Letting the base class behavior occur enables the derived class to concentrate on implementing behavior specific to the derived class. If the base class implementation is not called, it is up to the derived class to make their behavior compatible with the behavior of the base class.  
-  
-## In This Section  
-  
-- [Versioning with the Override and New Keywords](./versioning-with-the-override-and-new-keywords.md)  
-  
-- [Knowing When to Use Override and New Keywords](./knowing-when-to-use-override-and-new-keywords.md)  
-  
+> It is recommended that virtual members use `base` to call the base class implementation of that member in their own implementation. Letting the base class behavior occur enables the derived class to concentrate on implementing behavior specific to the derived class. If the base class implementation is not called, it is up to the derived class to make their behavior compatible with the behavior of the base class.
```

Comment:
Is there some prescriptive guidance on this we can link to? For example, is there a convention around naming a virtual/abstract member `OnDoSomething` you're indicating a callback for the purpose of being overridden and therefor you don't need to call `base.OnDoSomething`? I've always wondered that. I did a quick search but didn't find anything. Maybe you know..

Reply:
Andy, I don't know of any prescriptive guidance here.